plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,340 @@
+
+            xbmcgui.Dialog().notification("HTTP Status Error", error.message)
+        
+        index = 0
+
+        entries = []
+
+        for item in items:
+
+            #       Does not work: see https://forum.kodi.tv/showthread.php?tid=112916
+            #            if 'pubDate' in item:
+            #                pubDate = datetime.strptime(item['pubDate'][:-6], '%a, %d %b %Y %H:%M:%S')
+            #            else:
+            pubDate = None
+
+            entries = entries + [{
+                "path": str(index),
+                "name": item["title"],
+                "name2": item["description"] if "description" in item else "",
+                "icon": image,
+                "params": [
+                    {
+                        "call": "playRss",
+                        "index": str(index),
+                        "url": url
+                    }
+                ],
+                "pubDate": pubDate
+
+            }]
+
+            index += 1
+
+
+#        entries = sorted(entries, key=itemgetter('pubDate'), reverse=True)
+        for entry in entries:
+
+            self._add_list_item(entry, path)
+
+        xbmcplugin.endOfDirectory(self._addon_handle)
+
+    def play(self, parent, path, params):
+
+        url = params["url"][0]
+        xbmc.executebuiltin('PlayMedia(%s)' % url)
+
+    def _get_node_by_path(self, path):
+
+        if path == "/":
+            return self._menu[0]
+
+        tokens = path.split("/")[1:]
+        directory = self._menu[0]
+
+        while len(tokens) > 0:
+            path = tokens.pop(0)
+            for node in directory["node"]:
+                if node["path"] == path:
+                    directory = node
+                    break
+
+        return directory
+
+    def _build_param_string(self, params, current=""):
+
+        if params == None:
+            return current
+
+        for obj in params:
+            for name in obj:
+                current += "?" if len(current) == 0 else "&"
+                current += name + "=" + str(obj[name])
+
+        return current
+
+    def _add_list_item(self, entry, path):
+
+        if path == "/":
+            path = ""
+
+        item_path = path + "/" + entry["path"]
+        item_id = item_path.replace("/", "_")
+
+        param_string = ""
+        if "params" in entry:
+            param_string = self._build_param_string(entry["params"],
+                                                    current=param_string)
+
+        if "node" in entry:
+            is_folder = True
+        else:
+            is_folder = False
+
+        label = entry["name"]
+
+        if settings.getSetting("label%s" % item_id) != "":
+            label = settings.getSetting("label%s" % item_id)
+
+        if "icon" in entry and entry["icon"].startswith("http"):
+            icon_file = entry["icon"]
+
+        elif "icon" in entry:
+            icon_file = os.path.join(addon_dir,
+                                     "resources",
```

Comment:
It seems there is not "assets" folder in "resources"?

Reply:
Changed this so that there is now a default icon in case that no specific icon is given.

Reply:
I still think this will not work. As there are no extra files in the "assets" folder. I guess this should be icon.png?